### DISCUSSION TITLE
Channel emulator configuration

### DISCUSSION THREAD

**Discussion by fabio-patrizi** [srsran/srsRAN_Project#157] [General Help]
Hi all,
I'm trying to set up the channel emulator to introduce a delay. I need to get up to 600ms (NTN) but I started with a range of 80-90ms. I'm using srsRAN_4G 23.04 and srsRAN_Project 23.5, with zmq (old setting, didn't switch to current one yet).
The UE configuration file is as follows (channel section at the bottom).  

I do not observe any delay: ping in both directions reports a delay smaller than 80 ms. I have also tried with 400ms, expecting no communication to happen, but also in this case everything works as if no delay was introduced. 

Has anyone else used the channel emulator with zmq? Any help?

Thanks in advance!
Best,
F

[rf]
freq_offset = 0
tx_gain = 50
rx_gain = 40
srate = 11.52e6
nof_antennas = 1

device_name = zmq
device_args = tx_port=tcp://127.0.0.1:2001,rx_port=tcp://127.0.0.1:2000,base_srate=11.52e6

[rat.eutra]
dl_earfcn = 2850
nof_carriers = 0

[rat.nr]
bands = 3
nof_carriers = 1

[pcap]
enable = mac, mac_nr, nas
mac_filename = /tmp/ue_mac.pcap
mac_nr_filename = /tmp/ue_mac_nr.pcap
nas_filename = /tmp/ue_nas.pcap

[log]
all_level = debug
phy_lib_level = none
all_hex_limit = 32
filename = /tmp/ue.log
file_max_size = -1

[usim]
mode = soft
algo = milenage
opc  = 63BFA50EE6523365FF14C1F45F88737D
k    = 00112233445566778899aabbccddeeff
imsi = 001010123456780
imei = 353490069873319

[rrc]
release = 15
ue_category = 4

[nas]
apn = srsapn
apn_protocol = ipv4

[gw]
netns = ue1
ip_devname = tun_srsue
ip_netmask = 255.255.255.0

[gui]
enable = false

[channel.dl]
enable        = true
  
[channel.dl.delay]
enable        = true
period_s      = 10
init_time_s   = 0
maximum_us    = 80000
minimum_us    = 90000
 
[channel.ul]
enable        = true

[channel.ul.delay]
enable        = true
period_s      = 10
init_time_s   = 0
maximum_us    = 80000
minimum_us    = 90000

**Comment by pgawlowicz** [comment]
Hi, the channel emulator is only instantiated in the case of LTE/NSA UE. Currently, it is not supported for SA mode in the srsRAN_4G project.

**Reply by jjose001** [reply]
Hi, 

I followed this tutorial: https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#overview with zmq RU interfaces. Is the channel emulator still only instantiated on LTE case? Can I try different capabilities on srsUE with srsGNB as gNB? After changing some parameters, I dont see any change.

**Reply by pgawlowicz** [reply]
You can connect 5G srsUE and gnb using the zmq-based Radio devices, that emulate the transmission of the signal without imposing any modifications to the signal. 

The channel emulator (that allows changing signal SNR, delay, etc.) is not available in the 5g srsUE.

**Reply by Zplusless** [reply]
Hi @pgawlowicz，If my understanding was right, the "channel emulator" you mentioned above refers to the "channel.*" parameters in the srsUE config file. My question is, is it possible to change signal SNR, delay, etc. using a GNU radio broker?

**Reply by pgawlowicz** [reply]
please check here: https://github.com/srsran/srsRAN_Project/issues/757
